binary/typedb

```diff
@@ -17,7 +17,12 @@
 #
 
 # TypeDB global variables
-JAVA_BIN=java
+JAVA_HOME=${JAVA_HOME:-java}
+if [[ "$JAVA_HOME" != "java" ]]; then
+    JAVA_HOME="$JAVA_HOME/bin/java"
+fi
+echo "Using java executable at $JAVA_HOME"
```

Comment:
Can we remove this `echo` statement? While I see that it can indeed be useful especially when troubleshooting, it would be too verbose.

Reply:
seems reasonable; i removed it

Reply:
I think the logic is correct, but can we rewrite it into something like this:

```
# TypeDB global variables
JAVA_BIN="java"
if [[ ! -z "$JAVA_HOME" ]]; then
    JAVA_BIN="$JAVA_HOME/bin/java"
fi
[[ $(readlink $0) ]] && path=$(readlink $0) || path=$0
TYPEDB_HOME=$(cd "$(dirname "${path}")" && pwd -P)
...
```

In your approach, you're setting `JAVA_HOME` to point to the `java` binary rather than the directory.

Reply:
done :)